### PR TITLE
HDDS-3809. Make number of open containers on a datanode a function of no of volumes reported by it.

### PR DESCRIPTION
**Pull request by bshashikant**
## What changes were proposed in this pull request?
Added logic to take into account max no of healthy disks reported among the sets
of datanodes of a pipeline to get the total no of open containers per pipeline.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3809

## How was this patch tested?

Added unit tests.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java [hunk 1]
```diff
@@ -608,6 +608,16 @@ public void close() throws IOException {
     pipelineFactory.shutdown();
   }
 
+  /**
+   * returns max count of healthy volumes from the set of
+   * datanodes constituting the pipeline
```
Comment (dineshchitlangia) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java:613]:
```suggestion
   * Returns count of healthy volumes from the set of
   * datanodes constituting the pipeline.
```

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/MockNodeManager.java [hunk 2]
```diff
@@ -605,6 +616,12 @@ public void setNetworkTopology(NetworkTopology topology) {
       this.currentState = currentState;
     }
 
+    NodeData(long capacity, long used, long currentState, long healthyVolumes) {
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/MockNodeManager.java:619]: Parameter `healthyVolumes` is unused.

Comment (bshashikant) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/MockNodeManager.java]: Addressed in the latest commit.

**Comment by adoroszlai** [comment]
Thanks @bshashikant for the contribution, @elek and @dineshchitlangia for the review.  Merged to `master` after latest green run.

**Pull request merged**